### PR TITLE
Set grant option to true for all privileges in sentry and support getting roles for users

### PR DESCRIPTION
**Pull request by rohitsinha54**
Issue: https://issues.cask.co/browse/CDAP-9029 and https://issues.cask.co/browse/CDAP-9035

See issue for details

### CODE REVIEW

File: cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java [hunk 1]
```diff
@@ -429,8 +430,24 @@ public Void run(SentryGenericServiceClient client) throws Exception {
...
+          return client.listAllRoles(requestingUser, COMPONENT_NAME);
+        }
+        if (principal.getType().equals(Principal.PrincipalType.USER)) {
+          // for a user get all the groups and their roles
+          Set<String> groups = authProvider.getGroupMapping().getGroups(principal.getName());
+          Set<TSentryRole> roles = new HashSet<>();
+          for (String group : groups) {
+            roles.addAll(client.listRolesByGroupName(requestingUser, group, COMPONENT_NAME));
+          }
+          return roles;
```
Comment (bdmogal) [inline cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java]: better to make this immutable?

**Review by bdmogal — APPROVED** [review]
1 comment about making the list of roles immutable. Anyway to add a test for these situations?

Other than that, LGTM

**Comment by bdmogal** [comment]
anyway we can add a test for this?

**Comment by rohitsinha54** [comment]
There is a unit test which tests the model policy part from  static file. I think we investigating it before but you cannot start a db based Sentry which allows revoking/granting operation. The CDAP Authorization integration test is supposed to test all this not sure what it does. 

Although its very clear now that we need all these tests. We can work on writing an authorization test which does this.

**Comment by rohitsinha54** [comment]
@bdmogal : Merged the PR and have opened a jira to add tests for RBAC scenario as a blocker for 4.2 https://issues.cask.co/browse/CDAP-9047

**Pull request merged**